webapp-client/src/component/Navbar.js

```diff
@@ -1,7 +1,71 @@
+import allowErrorMessage from '../errorify';
 import logo from '../lobster-icon.jpg';
 import React from 'react';
 
 export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loggedIn: false,
+      authInstance: null,
+    };
+    allowErrorMessage(this);
+    this.formatUser = this.formatUser.bind(this);
+    this.gapiSetState = this.gapiSetState.bind(this);
+    this.setLoggedIn = this.setLoggedIn.bind(this);
+    this.signOut = this.signOut.bind(this);
+    this.onAuth2Init = this.onAuth2Init.bind(this);
+  }
+
+  componentDidMount() {
+    window.gapi.load('auth2', this.gapiSetState);
+  }
+
+  gapiSetState() {
+    window.gapi.auth2
+      .init({
+        client_id: process.env.REACT_APP_CLIENT_ID,
+      })
+      .then(
+        this.onAuth2Init,
+        // Called if there is an init error
+        this.showErrorMessage
+      );
+  }
+
+  onAuth2Init() {
+    const authInstance = window.gapi.auth2.getAuthInstance();
+    this.setState({authInstance, loggedIn: authInstance.isSignedIn.get()});
+    authInstance.isSignedIn.listen(this.setLoggedIn);
+  }
+
+  setLoggedIn(loggedIn) {
+    this.setState({loggedIn});
+  }
+
+  signOut() {
+    if (this.state.authInstance) {
```

Comment:
I realized we don't need this because if authInstance were missing, then it would have been impossible for them to sign in so then the Sign Out button wouldn't be showing anyway.

Reply:
Your comment made me think that it might be possible in a fringe case that the auth instance has expired? But I think in production it would still just reload and the behavior would be the same, so I'll take it back out.